### PR TITLE
Add Makefile

### PR DESCRIPTION
**Pull request by kkc**
[Why]
1. To make ppl easier to understand how to build, lint, and test this
project.
2. make workflow reproducible

[How]
1. Add Makefile
2. Add several stages including:
    - deps: Ensures fresh go.mod and go.sum.
    - lint: adopt golangci-lint to improve code quality
    - format: Formats Go code
    - test-unit: run unit tests
    - test-integration: run integration tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	@echo ">> formatting code"
+	@gofmt -s -w $(FILES_TO_FMT)
+
+.PHONY: test-unit
+test-unit: ## Runs all Youtube Go unit tests
+test-unit:
+	@go test -v
+
+.PHONY: test-integration
+test-integration: ## Runs all Youtube Go unit tests
```
Comment (Julian-Chu) [inline Makefile:48]: remove "unit"

**Comment by Julian-Chu** [comment]
Could you also put "gofmt -l ." in Makefile?  I think later we could integrate Github Actions to do format check for PR.

File: Makefile (hunk 1 above)
Comment (kkc) [reply Makefile:48]: good catch!

**Comment by kkc** [comment]
@Julian-Chu thx you for your comments. I think maybe we could use `golangci-lint` to perform the format check. I do see a lot of Go opensource projects using `golangci-lint` to catch such errors. how do you think?  

btw, I've enabled it in our Makefile. :)

**Comment by Julian-Chu** [comment]
great idea!  I didn't know golangci-lint before. It looks more powerful.

**Pull request merged**